web_src/js/modules/fetch.js

```diff
@@ -3,29 +3,33 @@ import {isObject} from '../utils.js';
 const {csrfToken} = window.config;
 
 // fetch wrapper, use below method name functions and the `data` option to pass in data
-// which will automatically set an appropriate content-type header. For json content,
-// only object and array types are currently supported.
-function request(url, {headers, data, body, ...other} = {}) {
+// which will automatically set an appropriate headers. For json content, only object
+// and array types are currently supported.
+export function request(url, {headers = {}, data, body, ...other} = {}) {
   let contentType;
   if (!body) {
     if (data instanceof FormData) {
-      contentType = 'multipart/form-data';
       body = data;
     } else if (data instanceof URLSearchParams) {
-      contentType = 'application/x-www-form-urlencoded';
       body = data;
     } else if (isObject(data) || Array.isArray(data)) {
       contentType = 'application/json';
       body = JSON.stringify(data);
     }
   }
 
+  const headersMerged = new Headers({
+    'x-csrf-token': csrfToken,
```

Comment:
It's strange to send CSRF token for GET requests.

Reply:
Yes, unless there are GETs in our code that need it. I think it's possible once we start to migrate all the ajax, but I guess those should be changed to POST then.

Reply:
Done in https://github.com/go-gitea/gitea/pull/27026/commits/7bcc2037540bf33674ee396c5a3024954c94f4e2.